### PR TITLE
[1.x] Bump jline3 to `3.27.0`

### LINKED ISSUE #7505: `illegal reflective access operation` using sbt 1.9.9 with Java 11 (but not 8 or 17 or 21)
* Does **not** occur with Java 17 and 21 (using sbt 1.9.9 as well), only shows on Java 11
* Does **not** occur with sbt 1.9.8

```
$ java -version
openjdk version "11.0.22" 2024-01-16
OpenJDK Runtime Environment Temurin-11.0.22+7 (build 11.0.22+7)
OpenJDK 64-Bit Server VM Temurin-11.0.22+7 (build 11.0.22+7, mixed mode)

$ cat project/build.properties 
sbt.version=1.9.9

$ sbt
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jline.terminal.impl.exec.ExecTerminalProvider$ReflectionRedirectPipeCreator (file:/home/mkurz/.sbt/boot/scala-2.12.18/org.scala-sbt/sbt/1.9.9/jline-terminal-3.24.1.jar) to constructor java.lang.ProcessBuilder$RedirectPipeImpl()
WARNING: Please consider reporting this to the maintainers of org.jline.terminal.impl.exec.ExecTerminalProvider$ReflectionRedirectPipeCreator
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[info] welcome to sbt 1.9.9 (Eclipse Adoptium Java 11.0.22)
...
```

### PR DESCRIPTION
**Pull request by Friendseeker**
Fixes https://github.com/sbt/sbt/issues/7505

c.c. https://discord.com/channels/632150470000902164/922600050989875282/1287082148037132378

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1176,8 +1176,15 @@ lazy val sbtClientProj = (project in file("client"))
     nativeImageReady := { () =>
       ()
     },
-    nativeImageVersion := "22.2.0",
-    nativeImageOutput := target.value / "bin" / "sbtn",
+    nativeImageVersion := "23.0",
+    nativeImageJvm := "graalvm-java23",
+    nativeImageOutput := {
+      val outputDir = (target.value / "bin").toPath
```
Comment (Friendseeker) [inline build.sbt:1182]: Without explicitly creating the dir there, the following Error is thrown

> Error: Writing image to non-existent directory /Users/runner/work/sbt/sbt/client/target/bin is not allowed. Create the missing directory if you want the image to be written to that location.

Example: https://github.com/Friendseeker/sbt/actions/runs/11008291362/job/30565658216

**Comment by Friendseeker** [comment]
@SethTisue I got the native image jobs to work and made this PR based on top of your work (#7500). Would you like to quickly take a look at this PR and see if things look alright?

The issue is that I know nothing about jline or sbt terminal implementation, so need someone with domain expertise like you to proceed with the PR.

**Comment by SethTisue** [comment]
I can have a look next week.

**Comment by SethTisue** [comment]
LGTM, not that I brought any particular expertise to bear on it 😁 

I guess what happened here is that JDK 23 went GA since I tried this, and so the trouble I hit simply doesn't exist anymore.

@eed3si9n wdyt?

I've submitted https://github.com/scala/scala/pull/10882 to scala/scala, and I'll do Scala 3 soon (UPDATE: here 'tis: https://github.com/scala/scala3/pull/21752)

**Review by eed3si9n — APPROVED** [review]
Yea. It would be good to fix that weird warning on JDK 11.

**Pull request merged**